x-pack/plugins/uptime/server/lib/requests/search/monitor_summary_iterator.ts

```diff
@@ -167,11 +166,6 @@ export class MonitorSummaryIterator {
 
   // Get a CursorPaginator object that will resume after the current() value.
   async paginationAfterCurrent(): Promise<CursorPagination | null> {
-    const peek = await this.peek();
-    if (!peek) {
-      return null;
-    }
-
```

Comment:
mainly this check is separate into a separate API, where instead of peeking which adds extra cost, we are doing a separate api call.